### PR TITLE
Implement smooth automap rotation

### PR DESCRIPTION
**Pull request by JNechaevsky**
This PR implements smooth rotation in automap rotate mode. 
Tested with capped framerate as well as with `-fliplevels`, everything fine.

### CODE REVIEW

File: src/doom/am_map.c [hunk 1]
```diff
@@ -1681,12 +1681,12 @@ static void AM_rotatePoint (mpoint_t *pt)
     pt->x -= mapcenter.x;
     pt->y -= mapcenter.y;
 
-    tmpx = (int64_t)FixedMul(pt->x, finecosine[mapangle>>ANGLETOFINESHIFT])
-         - (int64_t)FixedMul(pt->y, finesine[mapangle>>ANGLETOFINESHIFT])
+    tmpx = (int64_t)FixedMul(pt->x, finecosine[(ANG90 - viewangle)>>ANGLETOFINESHIFT])
```
Comment (fabiangreffrath) [inline src/doom/am_map.c:1684]: Could you turn this onto a `const angle_t actualangle = ANG90 - viewangle` instead of recalculating it four times, please?

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: src/doom/am_map.c (hunk 1 above)
Comment (JNechaevsky) [reply src/doom/am_map.c]: Sure, gladly. Done in https://github.com/fabiangreffrath/crispy-doom/pull/734/commits/fa66e7f1d17b752d1215f4b4544dca44fb360bf9.

**Comment by fabiangreffrath** [comment]
Awesome, thanks again!

**Comment by JNechaevsky** [comment]
My pleasure! I have two things to ask, please.

1. While automap rotation is now perfectly smooth, line movement is not really, especially comparing with PrBoom+. Is it because of pixels of map lines are moving using fixed, not floating point math? Not a critical necessary thing to implement (at least for now!), just wondering.
2. What's the real reason of using `const`? I know that `const` variables are never supposed to be changed, but looking down into `AM_rotatePoint` code I don't see any places where `viewangle` can be changed. So, is it a code philosophical, or right thing to do thing, or something else?

**Comment by fabiangreffrath** [comment]
Ad 1: honestly, I don't know. I have never really thought about automap interpolation before. 
Ad 2: Within the scope of this function, the variable is once initialized and then never changed again. The `const` keyword means exactly this and gives a *hint* to the compiler that this might be an optimization opportunity.

**Comment by JNechaevsky** [comment]
Situation with `const` now clear, thank you!
As about automap - just take a look how lines are moving in PrBoom+ when you'll have a minute, the difference will be quite notable, even if port resolution is set to 640x480.

**Comment by JNechaevsky** [comment]
May I ask for extra advice? I was wondering about `const`, and for example, can this replacement be considered as optimization? Or I'm on wrong track?

https://github.com/fabiangreffrath/crispy-doom/blob/master/src/doom/r_plane.c#L193
```
void R_ClearPlanes (void)
{
    int		i;
--- angle_t	angle;
+++ const angle_t	angle; // left to right mapping
    
    // opening / clipping determination
    for (i=0 ; i<viewwidth ; i++)
    {
	floorclip[i] = viewheight;
	ceilingclip[i] = -1;
    }

    lastvisplane = visplanes;
    lastopening = openings;
    
    // texture calculation
    memset (cachedheight, 0, sizeof(cachedheight));

--- // left to right mapping
--- angle = (viewangle-ANG90)>>ANGLETOFINESHIFT;
	
    // scale will be unit scale at SCREENWIDTH/2 distance
    basexscale = FixedDiv (finecosine[angle],centerxfrac);
    baseyscale = -FixedDiv (finesine[angle],centerxfrac);
}
```

_This is not for PR, just self-education._

**Comment by fabiangreffrath** [comment]
You can do this here and this would be a typical example. By declaring a variable as `const` you promise to the compiler that you will never change the variable value once it's initialized. This leaves it some more freedom to decide where to store that variable. However, nowadays most compilers should be smart enough to figure this out themselves. Also, this doesn't really optimize your code in terms of speed, but in terms of cleanliness and readability. 😉

**Comment by JNechaevsky** [comment]
Ah, perfectly clear now, thank you very much!

**Pull request merged**